In [1]:
import pandas as pd
import numpy as np
filename = '/content/drive/MyDrive/Week2_data/Machine Learning Data Sets/adult.csv'
df = pd.read_csv(filename)
df.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

**TARGET:income**

**What does one row represent? (A person? A business? An event? A product?)**

In [2]:
first_row = df.iloc[1]
first_row

age                                38
workclass                     Private
fnlwgt                          89814
education                     HS-grad
educational-num                     9
marital-status     Married-civ-spouse
occupation            Farming-fishing
relationship                  Husband
race                            White
gender                           Male
capital-gain                        0
capital-loss                        0
hours-per-week                     50
native-country          United-States
income                          <=50K
Name: 1, dtype: object

**The row above represent a person's profile in USA**

**How many features does the data have?**

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


**The dataset has over 14 features against the income target**

**How many rows are in the dataset?**

**The dataset as 48841 entries**

**What opportunities exist for dimensionality reduction or feature selection with this dataset?**

Yes, we can reduce education+eductional_num, marital_status+relationship+gender. capital_loss+capital_gain

In [4]:
# Convert the "educational-num" column to string type
df['educational-num'] = df['educational-num'].astype(str)

In [5]:
df['Family_status'] = df['marital-status'] + df['relationship'] + df['gender']
df['Asset_Value'] =df['capital-loss'] + df['capital-gain']
df['Education_status'] = df['education'] + df['educational-num']
df.head()

age  workclass  fnlwgt     education educational-num      marital-status  \
0   25    Private  226802          11th               7       Never-married   
1   38    Private   89814       HS-grad               9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm              12  Married-civ-spouse   
3   44    Private  160323  Some-college              10  Married-civ-spouse   
4   18          ?  103497  Some-college              10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income                  Family_status  \
0              40  United-States  <=50K     Never-marriedOwn-childMale   
1              50  United-States  <=50K  Married-civ-spouseHusbandMale   
2              40  United-States   >50K  Married-civ-spouseHusbandMale   
3              40  United-States   >50K  Married-civ-spouseHusbandMale   
4              30  United-States  <=50K   Never-marriedOwn-childFemale   

   Asset_Value Education_status  
0            0            11th7  
1            0         HS-grad9  
2            0     Assoc-acdm12  
3         7688   Some-college10  
4            0   Some-college10

In [6]:
df.drop(columns=['marital-status', 'relationship', 'gender', 'capital-loss', 'capital-gain'], inplace=True)
df.head()

age  workclass  fnlwgt     education educational-num         occupation  \
0   25    Private  226802          11th               7  Machine-op-inspct   
1   38    Private   89814       HS-grad               9    Farming-fishing   
2   28  Local-gov  336951    Assoc-acdm              12    Protective-serv   
3   44    Private  160323  Some-college              10  Machine-op-inspct   
4   18          ?  103497  Some-college              10                  ?   

    race  hours-per-week native-country income                  Family_status  \
0  Black              40  United-States  <=50K     Never-marriedOwn-childMale   
1  White              50  United-States  <=50K  Married-civ-spouseHusbandMale   
2  White              40  United-States   >50K  Married-civ-spouseHusbandMale   
3  Black              40  United-States   >50K  Married-civ-spouseHusbandMale   
4  White              30  United-States  <=50K   Never-marriedOwn-childFemale   

   Asset_Value Education_status  
0            0            11th7  
1            0         HS-grad9  
2            0     Assoc-acdm12  
3         7688   Some-college10  
4            0   Some-college10

In [8]:
df.head()

age  workclass  fnlwgt     education educational-num         occupation  \
0   25    Private  226802          11th               7  Machine-op-inspct   
1   38    Private   89814       HS-grad               9    Farming-fishing   
2   28  Local-gov  336951    Assoc-acdm              12    Protective-serv   
3   44    Private  160323  Some-college              10  Machine-op-inspct   
4   18          ?  103497  Some-college              10                  ?   

    race  hours-per-week native-country income                  Family_status  \
0  Black              40  United-States  <=50K     Never-marriedOwn-childMale   
1  White              50  United-States  <=50K  Married-civ-spouseHusbandMale   
2  White              40  United-States   >50K  Married-civ-spouseHusbandMale   
3  Black              40  United-States   >50K  Married-civ-spouseHusbandMale   
4  White              30  United-States  <=50K   Never-marriedOwn-childFemale   

   Asset_Value Education_status  
0            0            11th7  
1            0         HS-grad9  
2            0     Assoc-acdm12  
3         7688   Some-college10  
4            0   Some-college10

In [10]:
df.dtypes

age                  int64
workclass           object
fnlwgt               int64
education           object
educational-num     object
occupation          object
race                object
hours-per-week       int64
native-country      object
income              object
Family_status       object
Asset_Value          int64
Education_status    object
dtype: object

In [11]:
df.tail()

age     workclass  fnlwgt   education educational-num  \
48837   27       Private  257302  Assoc-acdm              12   
48838   40       Private  154374     HS-grad               9   
48839   58       Private  151910     HS-grad               9   
48840   22       Private  201490     HS-grad               9   
48841   52  Self-emp-inc  287927     HS-grad               9   

              occupation   race  hours-per-week native-country income  \
48837       Tech-support  White              38  United-States  <=50K   
48838  Machine-op-inspct  White              40  United-States   >50K   
48839       Adm-clerical  White              40  United-States  <=50K   
48840       Adm-clerical  White              20  United-States  <=50K   
48841    Exec-managerial  White              40  United-States   >50K   

                       Family_status  Asset_Value Education_status  
48837   Married-civ-spouseWifeFemale            0     Assoc-acdm12  
48838  Married-civ-spouseHusbandMale            0         HS-grad9  
48839         WidowedUnmarriedFemale            0         HS-grad9  
48840     Never-marriedOwn-childMale            0         HS-grad9  
48841   Married-civ-spouseWifeFemale        15024         HS-grad9